src/modules/communities/pages/CommunityPage/CommunityManagementPage/CommunityManagementPage.tsx

```diff
@@ -1,50 +1,67 @@
 import React, { useEffect, useState } from 'react';
-import { useParams } from 'react-router-dom';
+import { useHistory, useParams } from 'react-router-dom';
 import Grid from '@material-ui/core/Grid';
-import { PageTitle, Loading, CustomLabel, toast } from 'shared/components';
-import CustomCardWithIcon from 'shared/components/CustomCardWithIcon/CustomCardWithIcon';
+import {
+  PageTitle,
+  Loading,
+  CustomLabel,
+  toast,
+  CustomCardWithIcon,
+} from 'shared/components';
 import BikeService from 'modules/bicycles/services/BikeService';
 import AmountBikesPerCommunity from 'modules/bicycles/utils/AmountBikesPerCommunity';
 import CommunityService from 'modules/communities/services/CommunityService';
 import Community from 'modules/communities/models/Community';
 
+type Params = {
+  id: string;
+};
+
 const CommunityManagementPage: React.FC = () => {
-  const { id } = useParams<{ id: string }>();
   const [community, setCommunity] = useState<Community>();
   const [loading, setLoading] = useState<boolean>(false);
   const [amountsBikesPerCommunity, setAmountsBikesPerCommunity] =
     useState<AmountBikesPerCommunity>();
 
-  useEffect(() => {
-    if (id) {
-      setLoading(true);
-      CommunityService.getCommunityById(id)
-        .then(res => {
-          setCommunity(res);
-        })
-        .catch(() => {
-          toast.error('Erro ao carregar comunidade.');
-        })
-        .finally(() => {
-          setLoading(false);
-        });
+  const history = useHistory();
+  const { id: communityId } = useParams<Params>();
+
+  const showFeedbackError = () => {
+    toast.error('Erro ao carregar comunidade.');
+  };
+
+  const fetchData = async id => {
+    setLoading(true);
 
-      BikeService.getAmountFilteredBikesPerCommunity(id)
-        .then(res => {
-          setAmountsBikesPerCommunity(res);
-        })
-        .catch(() => {
-          toast.error('Erro ao carregar lista de comunidades.');
-        });
+    try {
+      const communityData = await CommunityService.getCommunityById(id);
+      const bikeData = await BikeService.getAmountFilteredBikesPerCommunity(id);
+
+      setCommunity(communityData);
+      setAmountsBikesPerCommunity(bikeData);
+    } catch {
+      showFeedbackError();
+    } finally {
+      setLoading(false);
     }
-  }, [id]);
+  };
+
+  useEffect(() => {
+    if (!communityId) return;
+    fetchData(communityId);
```

Comment:
O build ta reclamando desse `fetchData`. Entendi que ela é uma função que tu criou. Deve ter mais alguma coisa que deve se fazer aqui.

Reply:
` Line 52:6:  React Hook useEffect has a missing dependency: 'fetchData'. Either include it or remove the dependency array  react-hooks/exhaustive-deps`

Reply:
Será que é um mesmo nome de uma função do React?

Reply:
Hum.. boa. Vou resolver isso aqui :)